packages/main/src/Carousel.ts

```diff
@@ -549,19 +535,41 @@ class Carousel extends UI5Element {
 	}
 
 	get effectiveItemsPerPage(): number {
+		const itemsPerPageArray = this.itemsPerPage.split(" ");
+		let itemsPerPageSizeS = 1,
+			itemsPerPageSizeM = 1,
+			itemsPerPageSizeL = 1,
+			itemsPerPageSizeXL = 1;
+
+		itemsPerPageArray.forEach(element => {
+			if (element.startsWith("S")) {
+				itemsPerPageSizeS = Number(element.slice(1));
```

Comment:
Make it
itemsPerPageSizeS = Number(element.slice(1)) || 1;

This way if the value is bad - not a number, it will fallback to default 1.